backend/taxHandlers.ts

```diff
@@ -0,0 +1,259 @@
+      chargeTaxes: line.variant?.product.chargeTaxes || true,
+      taxCode: taxCode || DEFAULT_TAX_CODE,
+      quantity: line.quantity,
+      totalAmount: line.totalPrice.gross.amount,
+      itemCode: line.productSku || line.variant!.id,
+      description: line.productName,
+      discounted: order.discounts.length !== 0,
+    };
+  });
+  linesPayload.push({
+    id: "",
+    chargeTaxes: true,
+    taxCode: "FR000000",
+    quantity: 1,
+    totalAmount: order.shippingPrice.gross.amount,
+    discounted: false,
+    itemCode: "Shipping",
+    description: "",
+  });
+  return linesPayload;
+};
+const calculateTaxes = async (
+  taxData: TaxBaseSubscriptionFragment,
+  avataxConfig: AvataxConfig
+) => {
+  const lines = prepareLinesPayload(taxData);
+  const payload: FetchTaxesPayload = {
+    address: {
+      line1: taxData.address!.streetAddress1,
+      line2: taxData.address!.streetAddress2,
+      line3: "",
+      city: taxData.address!.city,
+      region: taxData.address!.countryArea,
+      country: taxData.address!.country.code,
+      postalCode: taxData.address!.postalCode,
+      locationCode: "",
+    },
+    discount:
+      taxData.discounts.reduce(
+        (total, discount) => total + discount.amount.amount,
+        0
+      ) || 0,
+    date: taxData.sourceObject.date,
+    currency: taxData.currency,
+    pricesEnteredWithTax: taxData.pricesEnteredWithTax,
+    lines: lines,
+  };
+  try {
+    const taxResponse =
+      lines.length !== 0
+        ? await fetchTaxes(payload, avataxConfig, "SalesOrder")
+        : undefined;
+    if (!taxResponse) {
+      const data: ResponseTaxPayload = {
+        shipping_price_gross_amount: String(taxData.shippingPrice.amount),
+        shipping_price_net_amount: String(taxData.shippingPrice.amount),
+        shipping_tax_rate: "0",
+        lines: taxData.lines.map((line) => ({
+          tax_rate: "0",
+          total_gross_amount: String(line.totalPrice.amount),
+          total_net_amount: String(line.totalPrice.amount),
+        })),
+      };
+      return { body: { data: data }, status: 200 };
+    }
+    const shipping = taxResponse.lines.find(
+      (line) => line.itemCode === "Shipping"
+    );
+    const shippingDiscount = shipping?.discountAmount || 0;
+    const shippingNet =
+      (shipping?.lineAmount
+        ? shipping.lineAmount
+        : taxData.shippingPrice.amount) - shippingDiscount;
+    const shippingTax = shipping?.tax ? shipping.tax : 0;
+    const taxRate = shipping?.details.reduce(
+      (currentRate, detail) => currentRate + (detail?.rate || 0),
+      0
+    );
+    const data: ResponseTaxPayload = {
+      shipping_price_gross_amount: (shippingNet + shippingTax).toFixed(2),
+      shipping_price_net_amount: shippingNet.toFixed(2),
+      // Tax rate send to Saleor should be in decimal.
+      // when tax rate is 23%, the value that we send should be
+      // taxRate = 23
+      shipping_tax_rate: ((taxRate || 0) * 100).toFixed(2),
```

Comment:
This can be extracted into a separate helper function (it's used in multiple places)